plugins/module_utils/network/sonic/config/radius_server/radius_server.py

```diff
@@ -331,115 +322,279 @@ def get_radius_server_payload(self, hosts):
+        requests.extend(self.get_delete_servers(commands))
 
         return requests
+
+    @staticmethod
+    def sort_lists_in_config(config):
+        """This method sorts the lists in the radius server configuration"""
+        if config and config.get('servers') and config['servers'].get('host'):
+            config['servers']['host'].sort(key=lambda x: x['name'])
+
+    def remove_default_entries(self, data):
+        """This method removes default entries from the radius server configuration"""
+        if data:
+            if data.get('auth_type') == 'pap':
+                data.pop('auth_type')
+            if data.get('timeout') == 5:
+                data.pop('timeout')
+            if data.get('servers') and data['servers'].get('host'):
+                pop_list = []
+                for idx, host in enumerate(data['servers']['host']):
+                    if len(host) == 1:
+                        continue
+                    if host.get('protocol') == 'UDP':
+                        host.pop('protocol')
+                    if host.get('port') == 1812:
+                        host.pop('port')
+                    if len(host) == 1:
+                        pop_list.insert(0, idx)
+
+                for idx in pop_list:
+                    data['servers']['host'].pop(idx)
+
+                if not data['servers']['host']:
+                    data.pop('servers')
+
+    def get_replaced_config(self, want, have):
+        """This method returns the radius server configuration to be replaced and the respective delete requests"""
+        config_dict = {}
+        requests = []
+        cp_want = deepcopy(want)
+        cp_have = deepcopy(have)
+        self.remove_default_entries(cp_want)
+        self.remove_default_entries(cp_have)
+        self.sort_lists_in_config(cp_want)
+        self.sort_lists_in_config(cp_have)
+
+        if not cp_want or not cp_have:
+            return config_dict, requests
+
+        # Handle host level replacement
+        if cp_want.get('servers') and len(cp_want) == 1:
+            if not cp_have.get('servers'):
+                return config_dict, requests
+
+            cfg_host_dict = {host['name']: host for host in cp_have['servers']['host']}
+            host_list = []
+            for host in cp_want['servers']['host']:
+                name = host['name']
+                cfg_host = cfg_host_dict.get(name)
+
+                if not cfg_host:
+                    continue
+                if host != cfg_host:
+                    url = f'{RADIUS_SERVER_PATH}/servers/server={cfg_host["name"]}'
+                    requests.append({'path': url, 'method': DELETE})
+                    host_list.append(cfg_host)
+            if host_list:
+                config_dict['servers'] = {'host': host_list}
+
+        # Handle config top level replacement
+        elif cp_want != cp_have:
+            requests.append({'path': RADIUS_SERVER_PATH, 'method': DELETE})
```

Comment:
Generate delete requests similar to delete all scenario.

Reply:
Addressed.